vm/src/emulator/executor.rs

```diff
@@ -1233,9 +1232,17 @@ impl Emulator for LinearEmulator {
             .collect();
         initial_memory.extend(public_input_iter);
 
-        let tracked_ram_size = self
-            .memory_layout
-            .tracked_ram_size(self.initial_static_ram_image.len_bytes() + rom_initialization.len());
+        let tracked_ram_size = self.memory_layout.tracked_ram_size(
+            self.initial_static_ram_image.len_bytes()
+                + match self.static_rom_image_index {
+                    None => 0,
+                    Some(uidx) => self
```

Comment:
Would it be possible to do this count the first time through (https://github.com/nexus-xyz/nexus-zkvm/pull/398/files#diff-ac93bf3e647b4debf9b9fe14b595fb6ca897150133c8edc1fd172d4edfd4a883R1191-R1198) and save it to use here, rather than doing the full iteration again?